omod/src/main/java/org/openmrs/module/idgen/rest/resource/LogEntryResource.java

```diff
@@ -0,0 +1,151 @@
+        Date dateFrom = fromDate != null ? (Date) ConversionUtil.convert(fromDate, Date.class) : null;
+        Date dateTo = toDate != null ? (Date) ConversionUtil.convert(toDate, Date.class) : null;
+        User user = generatedBy != null ? service.getUserByUuid(generatedBy) : null;
+        if (logSource == null && identifier == null && comment == null && dateFrom == null && dateTo == null
+                && user == null) {
+            return new EmptySearchResult();
+        }
+        logEntries = identifierSourceService.getLogEntries(logSource, dateFrom, dateTo, identifier, user, comment);
+
+        return new NeedsPaging<LogEntry>(logEntries, context);
+    }
+
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+        DelegatingResourceDescription description = null;
+        if (rep instanceof RefRepresentation) {
+            description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("display");
+            description.addProperty("identifier");
+            description.addSelfLink();
+        } else if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+            description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("source");
+            description.addProperty("identifier");
+            description.addProperty("comment");
+            description.addProperty("generatedBy");
+            description.addProperty("dateGenerated");
+            description.addSelfLink();
+            if (rep instanceof DefaultRepresentation) {
+                description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            }
+        }
+        return description;
+    }
+
+    @PropertyGetter("uuid")
+    public String getUuid(LogEntry logentry) {
+        return logentry.getIdentifier();
+    }
+
+    @PropertyGetter("source")
+    public String getSource(LogEntry logentry) {
+        return logentry.getSource().getUuid();
+    }
+
+    @PropertyGetter("generatedBy")
+    public String getGeneratedBy(LogEntry logentry) {
+        return logentry.getGeneratedBy().getUuid();
+    }
+
+    @PropertyGetter("display")
+    public String getDisplay(LogEntry logentry) {
+        return logentry.getSource().toString();
+    }
+
+    @Override
+    public LogEntry newDelegate() throws ResourceDoesNotSupportOperationException {
+        throw new UnsupportedOperationException("method not supported");
+    }
+
+    @Override
+    public LogEntry save(LogEntry delegate) throws ResourceDoesNotSupportOperationException {
+        throw new UnsupportedOperationException("save not supported");
+
+    }
+
+    @Override
+    public LogEntry getByUniqueId(String uniqueId) {
+        return Context.getService(IdentifierSourceService.class).getLogEntries(null, null, null, uniqueId, null, null).get(0);
```

Comment:
According to the getUuid() method that we added to LogEntry, the uniqueId should be taken to mean the identifier. So this should fetch the corresponding log entry instead of the hard coded get(0)

Reply:
getLogEntries returns a list of log entries but since I have set the identifier to be equal to the uniqueId, it will return a one item list so unless I use the get(0), I get a Type mismatch error. How do I go about this?

Reply:
Do you mean that there is no method to get log entries that match a given identifier?

Reply:
In the identifier source service, there is no other method for fetching log entries except getLogEntries()

Reply:
Oh i see! So what happens if no entry is found that matches the given uniqueId?

Reply:
not sure how thats handled

Reply:
If the list size is zero, .get(0) will throw an IndexOutOfBounds exception. Were you aware of that?

Reply:
Oh yes, but then getLogEntries() returns a list. Can I maybe use and if statement and have that return statement for when the list size isnt 0?

Reply:
Another lesson for you to learn. You do not need to ask for permission to try out anything. Always do what you think is best and wait for a review. 😊

Reply:
Noted for the future